lib/str-utils.c

```diff
@@ -60,3 +60,9 @@ __normalize_key(const gchar *buffer)
 {
   return str_replace_char(buffer, '-', '_');
 }
+
+gchar *
+normalize_flag(const gchar *buffer)
+{
+  return str_replace_char(buffer, '_', '-');
```

Comment:
I had an earlier version when I tried to use `__normalize_key`, but I ran into all kind of problems, the code broke in unexpected ways, that's why ended up with the opposite normalization.

The thing is syslog-ng already has `cfg_process_flag` which is used to normalize flags. Also `__normalize_key` is not used for flags, but for block arguments, so it would be more important to align to `cfg_process_flag` instead of `__normalize_key`.

`cfg_process_flag` works differently than `__normalize_key`, it transforms underscores to hyphen. In my first attempt, I tried to use `__normalize_key` for every cases, even in `cfg_process_flag`. But the patch complexity increased quickly, and unit tests broke unexpectedly, as it turned out `cfg_process_flag` is used not only for flags, but in some other cases, for example template function command line arguments that depend on value pairs: like format-json. You can follow this by starting from `cfg_process_flag` -> `value_pairs_add_scope` -> `vp_cmdline_parse_scope` -> `value_pairs_new_from_cmdline` -> references.

In summary, the normalization from underscore and hyphen is more dominant in the current core philosophy, and I turned to investigate if I can align the exceptions to `cfg_process_flag`. This resulted in a much simpler patch.

Then I investigated, if I could use `cfg_process_flag` everywhere thus avoiding introducing a new function. It turned out it is not easy either: `cfg_process_flag` works by side effects (setting flag into a memory area), and in some cases we need explicit lookup function without side effects (check log_expr_node_lookup_flag("final") in cfg-tree.c).

Still I could have rewriten some cases to use `cfg_process_flag`, but its API is somewhat complex, and I could have not done it without severe complexity on the patch. I did not feel the the benefit to do so.

The only question that remains, if we can align the block argument normalization and flag normalization: both use normalizing from underscore to hyphen. Maybe it is possible, it can be investigated in another day.

Reply:
Thanks for the explanation, I can agree with you to move this PR forward as is.